frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -144,89 +151,147 @@ interface Interval {
+ * @returns The timestamp in seconds.
  */
-export function convertToSeconds(
-  value: number | bigint,
+function convertTimestampToSeconds(
+  timestamp: number | bigint,
   unit: number
 ): number {
   let unitAdjustment
 
-  if (unit === 1) {
+  if (unit === TimeUnit.MILLISECOND) {
     // Milliseconds
     unitAdjustment = 1000
-  } else if (unit === 2) {
+  } else if (unit === TimeUnit.MICROSECOND) {
     // Microseconds
     unitAdjustment = 1000 * 1000
-  } else if (unit === 3) {
+  } else if (unit === TimeUnit.NANOSECOND) {
     // Nanoseconds
     unitAdjustment = 1000 * 1000 * 1000
   } else {
     // Interpret it as seconds as a fallback
-    return Number(value)
+    return Number(timestamp)
   }
 
   // Do the calculation based on bigints, if the value
   // is a bigint and not safe for usage as number.
   // This might lose some precision since it doesn't keep
   // fractional parts.
-  if (typeof value === "bigint" && !Number.isSafeInteger(Number(value))) {
-    return Number(value / BigInt(unitAdjustment))
+  if (
+    typeof timestamp === "bigint" &&
+    !Number.isSafeInteger(Number(timestamp))
+  ) {
+    return Number(timestamp / BigInt(unitAdjustment))
   }
 
-  return Number(value) / unitAdjustment
+  return Number(timestamp) / unitAdjustment
 }
 
-function formatTime(data: number, field?: Field): string {
-  const timeInSeconds = convertToSeconds(data, field?.type?.unit ?? 0)
-  return moment
-    .unix(timeInSeconds)
+/**
+ * Converts a timestamp to a date object.
+ *
+ * @param timestamp The timestamp to convert.
+ * @param field The field containing the unit information.
+ * @returns The date object.
+ */
+export function convertTimestampToDate(
+  timestamp: number | bigint,
+  field?: Field
+): Date {
+  // Time values from arrow are not converted to a shared unit and
+  // just return the raw arrow value. Therefore, we need to adjust
+  // the value to seconds based on the unit information in the field.
+  // https://github.com/apache/arrow/blob/9e08c57c0986531879aadf7942998d26a94a5d1b/js/src/visitor/get.ts#L193C7-L209
+  const timeInSeconds = convertTimestampToSeconds(
+    timestamp,
+    field?.type?.unit ?? TimeUnit.SECOND
+  )
+  return moment.unix(timeInSeconds).utc().toDate()
+}
+
+/**
+ * Formats a time value based on the unit information in the field.
+ *
+ * @param timestamp The time value to format.
+ * @param field The field containing the unit information.
+ * @returns The formatted time value.
+ */
+function formatTime(timestamp: number | bigint, field?: Field): string {
+  const date = convertTimestampToDate(timestamp, field)
+  return moment(date)
     .utc()
-    .format(timeInSeconds % 1 === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
+    .format(date.getMilliseconds() === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
 }
 
-function formatDuration(data: number | bigint, field?: Field): string {
+function formatDate(date: number | Date): string {
```

Comment:
I think this is a little bit confusing because if you pass a `number` to this function, the returned String is simply that number and has nothing to do with the date format. Should the argument simply accept a `Date` and if the type is different, an error is thrown?